### PR TITLE
Feature/custom script and reduce

### PR DESCRIPTION
**Pull request by brnleehng**
Features
- Added custom scripts for users on booting up the nodes
- Set Reduce on a task with more than 1 chunk size. This allows the user to apply a reduce function on the current list of results in one task

- Added parallelThread support for uploading the environment if the environment is large (Not sure if it's necessary) The user should provide a resource file so the environment doesn't have to be downloaded on each job run.

### CODE REVIEW

**Comment by jiata** [comment]
is custom script backwards compatible?

File: R/utility.R [hunk 1]
```diff
@@ -416,3 +416,6 @@ createOutputFile <- function(filePattern, url){
   output
 }
 
+getSasTokenUrl <- function(){
```
Comment (brnleehng) [inline R/utility.R]: Removed.

File: R/doAzureParallel.R [hunk 2]
```diff
@@ -51,11 +51,33 @@ workers <- function(data){
...
-  if(!is.numeric(value)) stop("setChunkSize requires a numeric argument")
+  if (!is.numeric(value)) stop("setChunkSize requires a numeric argument")
 
   value <- max(round(value), 1)
 
-  assign("chunkSize", value, envir=.doAzureBatchGlobals)
+  assign("chunkSize", value, envir = .doAzureBatchGlobals)
+}
+
+#' Apply reduce function on a groups iterations of the foreach loop together per task.
```
Comment (zfengms) [inline R/doAzureParallel.R]: either "a group" or "groups"?

Comment (brnleehng) [inline R/doAzureParallel.R]: Fixed

**Comment by jiata** [comment]
LGTM

**Pull request merged**